commit 1b46c802c4193b5f7b39f55d305c00d546bd7e8b (refs/changes/58/4758/3)
Author: Rui Loura <rui@joyent.com>
Date:   2018-08-29T13:22:31+00:00 (1 year, 1 month ago)
    
    TRITON-722 "sdcadm health" still assumes each CN will have an "admin" nictag
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index eb6fb7e..258ff14 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,10 @@
 
 # sdcadm Changelog
 
+## 1.21.4
+
+- TRITON-722 "sdcadm health" still assumes each CN will have an "admin" nictag
+
 ## 1.21.3
 
 - TRITON-723 Improve the output of unhealthy instances from `sdcadm health`.
diff --git a/lib/common.js b/lib/common.js
index b495961..cb2d66c 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -1760,10 +1760,17 @@ function safeCycles() {
 
 function serverAdminIpFromSysinfo(sysinfo) {
     var nics = sysinfo['Network Interfaces'] || {};
-    var adminIp = Object.keys(nics).map(function (nicName) {
+    var tag = 'admin';
+    var adminIp;
+
+    if (sysinfo['Admin NIC Tag']) {
+        tag = sysinfo['Admin NIC Tag'];
+    }
+
+    adminIp = Object.keys(nics).map(function (nicName) {
         return nics[nicName];
     }).filter(function (nic) {
-        return nic['NIC Names'].indexOf('admin') !== -1;
+        return nic['NIC Names'].indexOf(tag) !== -1;
     }).map(function (nic) {
         return nic.ip4addr;
     })[0];
diff --git a/package.json b/package.json
index 24c9bb9..ea136e2 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a SmartDataCenter (SDC) standup",
-  "version": "1.21.3",
+  "version": "1.21.4",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
